src/v/coproc/partition.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/partition.h"
+#include "storage/fwd.h"
+
+#include <seastar/core/gate.hh>
+
+namespace coproc {
+
+class partition {
+public:
+    partition(storage::log, ss::lw_shared_ptr<cluster::partition>) noexcept;
+
+    ss::future<> start() { return ss::now(); }
+    ss::future<> stop() { return _gate.close(); }
+
+    model::offset start_offset() const { return _log.offsets().start_offset; }
+    model::offset last_offset() const { return _log.offsets().dirty_offset; }
```

Comment:
what the `last_offset` is ? Maybe it would be better to stick to the offset names that we already have instead of introducing new one ?

Reply:
good call